### PR TITLE
Added ability to group invoices by PIs

### LINKED ISSUE #7: Add post invoice script automation script to break Invoice into groups by PI
## Motivation
Regularly let users know how many resources they have used prior to receiving their official invoice.

## Completion Criteria
Invoice CSV grouped into groups by PI

## Description
- [ ] Create a way to group data in the invoice CSV by PI (individual CSVs or a database whatever is easier, developer's prerogative)
- [ ] This will then be used to email out this information but that is a different issue

## Completion dates
Desired - 2024-02-14
Required - TBD

### PR DESCRIPTION
**Pull request by QuanMPhm**
Closes #7, I have modified `process_report.py` to group the merged invoices into ones grouped by PIs. These smaller CSVs will be placed in a folder of your choice, or defaults to a one placed in the working directory

### CODE REVIEW

File: process_report/process_report.py [hunk 1]
```diff
@@ -53,8 +60,9 @@ def main():
 
     projects = list(set(projects + timed_projects_list))
 
-    remove_non_billables(merged_dataframe, pi, projects, args.output_file)
+    billable_projects = remove_non_billables(merged_dataframe, pi, projects, args.output_file)
```
Comment (naved001) [inline process_report/process_report.py:63]: can we still print the whole billable csv?

File: process_report/process_report.py [hunk 2]
```diff
@@ -53,8 +60,9 @@ def main():
...
     remove_billables(merged_dataframe, pi, projects, "non_billable.csv")
+    export_pi_billables(billable_projects, args.output_folder)
```
Comment (naved001) [inline process_report/process_report.py:65]: can we also make this optional and triggered by a flag? I think the default behavior should still be to produce the 2 reports i.e. billable and non-billable, but the user can say `--export-by-pi` and then it should split it out into multiple files. @joachimweyl thoughts?

File: process_report/process_report.py (hunk 1 above)
Comment (QuanMPhm) [reply process_report/process_report.py:63]: @naved001 I only modified `remove_non_billable` to return the dataframe with billable projects, so the function should still write out the billable csv

Comment (naved001) [reply process_report/process_report.py:63]: oh, right. I missed it, thanks!

File: process_report/process_report.py (hunk 2 above)
Comment (joachimweyl) [reply process_report/process_report.py:65]: that makes sense, as long as it is a flag we know to use and use every month until we decide to be done with it.

Comment (naved001) [reply process_report/process_report.py:65]: @QuanMPhm pointed out that the billable csv is still written, so I am okay with the current behavior then and don't care much about the flag.

Comment (QuanMPhm) [reply process_report/process_report.py:65]: @naved001 @joachimweyl It should only take minutes for me to make it optional for the script to export the invoice csv by pi or as combined or as both. I'll do what you prefer.

**Review by naved001 — APPROVED** [review]

**Pull request merged**